api/src/main/java/graphql/nadel/Service.java

```diff
@@ -45,4 +56,19 @@ public ServiceDefinition getServiceDefinition() {
     public DefinitionRegistry getDefinitionRegistry() {
         return definitionRegistry;
     }
+
+    public Object getServiceContext() {
+        return serviceContext;
+    }
+
+    public Service withContext(Object serviceContext) {
```

Comment:
I don't think this belongs here. A `Service` is something static for a given schema. I don't think we should change the lifecycle of it.

It also gets confusing because we pass in a `Map<String, Service>` i.e. all of the services to `NadelTransform.isApplicable`. It's weird having some `Service`s have a context and some that don't.

Also, in the future I'm looking to get rid of `Service` from the engine code once we have a combined runtime schema in place.

I would suggest moving this to `ExecutionContext.serviceContext` for now.

Reply:
Great suggestion! 
`NadelExecutionContext` is also already being passed down to transformers, which is handy.